### PR TITLE
removeMoney repair

### PR DESCRIPTION
**Pull request by sajrusvirus**
proposition for #1706

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 include(cotire)
 
 # Initialize CXXFLAGS.
-set(CMAKE_CXX_FLAGS                "-Wall -Werror -pipe")
+set(CMAKE_CXX_FLAGS                "-pedantic -Wall -Werror -pipe")
```
Comment (ranisalt) [inline CMakeLists.txt]: Why?

File: src/game.cpp [hunk 2]
```diff
@@ -1423,16 +1423,18 @@ bool Game::removeMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*
...
 	std::vector<Container*> containers;
 
-	std::multimap<uint64_t, Item*> moneyMap;
+	typedef std::multimap<uint64_t, Item*, std::less<uint64_t>> MoneyMap;
+	typedef MoneyMap::value_type moneymap_pair;
+	MoneyMap moneyMap;
 	uint64_t moneyCount = 0;
 
-	for (size_t i = cylinder->getFirstIndex(), j = cylinder->getLastIndex(); i < j; ++i) {
+	for (int32_t i = cylinder->getFirstIndex(), j = cylinder->getLastIndex(); i < j; ++i) {
```
Comment (ranisalt) [inline src/game.cpp]: Indexes must be `size_t`, revert it.

File: src/game.cpp [hunk 3]
```diff
@@ -1446,12 +1448,9 @@ bool Game::removeMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*
...
 			containers.push_back(container);
-		} else {
-			int32_t worth = item->getWorth();
-			if (worth != 0) {
-				moneyCount += worth;
-				moneyMap.emplace(worth, item);
-			}
+		} else if (item->getWorth() != 0) {
+			moneyCount += item->getWorth();
+			moneyMap.insert(moneymap_pair(item->getWorth(), item));
```
Comment (ranisalt) [inline src/game.cpp]: This way you just call a function twice instead of once, it's a drawback.

File: src/game.cpp [hunk 4]
```diff
@@ -1462,35 +1461,33 @@ bool Game::removeMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*
...
 				containers.push_back(tmpContainer);
-			} else {
-				int32_t worth = item->getWorth();
-				if (worth != 0) {
-					moneyCount += worth;
-					moneyMap.emplace(worth, item);
-				}
+			} else if (item->getWorth() != 0) {
+				moneyCount += item->getWorth();
+				moneyMap.insert(moneymap_pair(item->getWorth(), item));
```
Comment (ranisalt) [inline src/game.cpp]: Same as above

File: src/game.cpp [hunk 5]
```diff
@@ -1462,35 +1461,33 @@ bool Game::removeMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*
...
 	if (moneyCount < money) {
 		return false;
 	}
 
-	for (const auto& moneyEntry : moneyMap) {
-		Item* item = moneyEntry.second;
-		if (moneyEntry.first >= money) {
-			uint32_t worth = moneyEntry.first / item->getItemCount();
-			uint32_t removeCount = (money / worth) + 1;
+	for (MoneyMap::const_iterator mit = moneyMap.begin(), mend = moneyMap.end(); mit != mend && money > 0; ++mit) {
```
Comment (ranisalt) [inline src/game.cpp]: Why replace the range based for with the absolutely exact same, yet less semantic indexed for?

**Comment by ranisalt** [comment]
You added an entire copy of game.cpp to the commit. For your life, NEVER `git add -A` or `git add .` before `git status`, or even better, always `git status` before anything.

File: CMakeLists.txt (hunk 1 above)
Comment (sajrusvirus) [inline CMakeLists.txt]: sorry, it is just my habit

File: src/game.cpp (hunk 3 above)
Comment (sajrusvirus) [inline src/game.cpp]: yes, only 1453 line is necessary in this case

**Comment by sajrusvirus** [comment]
sorry form -A flag,

File: src/game.cpp (hunk 5 above)
Comment (sajrusvirus) [inline src/game.cpp]: hymm... but i think now i a bit faster, 2 comparison + 2 divide vs. 3 comparison.

**Comment by marksamman** [comment]
This is too much of a mess, it seems like you've just copied over parts of the function from a working copy from another source code, without proper understanding of what you've copied or what actually fixes the issue.

**Comment by sajrusvirus** [comment]
good reason to close.

**Comment by ranisalt** [comment]
Gosh he deleted his account :O

**Comment by marksamman** [comment]
That's unfortunate, I did not mean to discourage him/her from contributing.

This was a critical issue that had to be addressed ASAP, the quality of this PR was very low, the commit messages had multiple misspellings, it included a copy of game.cpp, and it was apparent that the author did not understand the changes (e.g. changing emplace to insert, and checking if an unsigned type is lesser than zero). It made no sense to keep the PR open when I had already fixed the issue.

**Pull request closed without merging**